frontend/packages/console-plugin-sdk/src/useExtensions.ts

```diff
@@ -40,47 +34,46 @@ import { Extension, ExtensionTypeGuard } from './typings';
  * };
  * ```
  *
- * @param typeGuard Type guard used to narrow the extension type.
+ * @param typeGuards Type guard(s) used to narrow the extension instances.
+ *
+ * @returns List of extension instances which are currently in use, narrowed by the
+ * given type guard(s).
  */
-export const useExtensions: UseExtensions = (typeGuard) => {
-  const allExtensions = pluginStore.getAllExtensions();
+export const useExtensions = <E extends Extension>(...typeGuards: ExtensionTypeGuard<E>[]): E[] => {
+  if (typeGuards.length === 0) {
+    throw new Error('You must pass at least one type guard to useExtensions');
+  }
 
-  // 1) Narrow extensions according to type guard
-  const matchedExtensions = React.useMemo(() => allExtensions.filter(typeGuard), [
-    allExtensions,
-    typeGuard,
-  ]);
+  const unsubscribeRef = React.useRef<() => void>(null);
```

Comment:
you could also use `VoidFunction`
```suggestion
  const unsubscribeRef = React.useRef<VoidFunction>(null);
```

but whatever you prefer, I dont mind

Reply:
Nice, I'll replace `() => void` with `VoidFunction` throughout the plugin SDK packages.